SU2_CFD/src/numerics/flow/convection/fds.cpp

```diff
@@ -157,9 +163,15 @@ CNumerics::ResidualType<> CUpwFDSInc_Flow::ComputeResidual(const CConfig *config
 
   MeandRhodT = 0.0; dRhodT_i = 0.0; dRhodT_j = 0.0;
   if (variable_density) {
-    MeandRhodT = -MeanDensity/MeanTemperature;
-    dRhodT_i   = -DensityInc_i/Temperature_i;
-    dRhodT_j   = -DensityInc_j/Temperature_j;
+    if (energy_multicomponent) {
+      MeandRhodT = -MeanDensity / (MeanCp * MeanTemperature);
+      dRhodT_i = -DensityInc_i / (Cp_i * Temperature_i);
+      dRhodT_j = -DensityInc_j / (Cp_j * Temperature_j);
+    } else {
+      MeandRhodT = -MeanDensity / MeanTemperature;
+      dRhodT_i = -DensityInc_i / Temperature_i;
+      dRhodT_j = -DensityInc_j / Temperature_j;
```

Comment:
Right, I'm going to stop reviewing now until you document your code properly

Reply:
that is why I put [WIP] in the pull request.

Reply:
The purpose of WIP is to get early feedback to avoid re-doing work later on. For people to give feedback some amount of documentation is necessary.

Reply:
OK, I am going to close the pull request